### PR TITLE
fix(stream): drop leaked chat bootstrap chunk for responses clients

### PR DESCRIPTION
**Pull request by CitrusIce**
## Summary
- drop leaked empty `chat.completion.chunk` bootstrap frames when the downstream client expects native Responses SSE
- keep the filter narrowly scoped to assistant-role empty bootstrap chunks so normal chat-completions traffic is untouched
- add regression coverage showing valid `response.*` events still pass through after the leaked frame is removed

## Context
Issue #186 made Responses passthrough format-aware for valid `response.*` payloads, but a stricter client can still fail when the stream begins with an empty Chat Completions bootstrap chunk before the first `response.created` event.

That frame is invalid on `/v1/responses` and breaks strict clients like OpenCode even though the rest of the stream is valid Responses SSE.

In practice this surfaces as schema/type-validation failure on the first SSE frame because the client expects `response.*` events only and instead receives `object: "chat.completion.chunk"`.

## Repro
A `/v1/responses` stream can begin like this:

```json
{
  "id": "chatcmpl-dummy",
  "object": "chat.completion.chunk",
  "choices": [
    {
      "index": 0,
      "delta": {
        "role": "assistant",
        "content": ""
      },
      "finish_reason": null
    }
  ]
}
```

followed by normal `response.created`, `response.in_progress`, and other `response.*` events.

Clients like OpenCode then fail on the first frame before they can consume the rest of the valid Responses stream.

## Why this fix
The existing Responses passthrough guard only activates once the payload is already recognizable as a Responses event (`parsed.type.startsWith("response.")`). This patch handles the residual edge case where the upstream leaks a single empty Chat Completions bootstrap chunk before the first valid Responses event.

## Testing
- `node --import tsx/esm --test tests/unit/stream-utils.test.ts`

### CODE REVIEW

File: open-sse/utils/stream.ts [hunk 1]
```diff
@@ -1255,6 +1255,35 @@ export function createSSEStream(options: StreamOptions = {}) {
...
+                        ? candidate.delta
+                        : null;
+
+                    if (!delta || delta.role !== "assistant") return false;
+                    if (!(delta.content === "" || delta.content === undefined)) return false;
+                    if (candidate.finish_reason !== null && candidate.finish_reason !== undefined) {
+                      return false;
+                    }
+
+                    return Object.keys(delta).every((key) => key === "role" || key === "content");
```
[the remark this answers is not in the document]
Comment (CitrusIce) [reply open-sse/utils/stream.ts]: Good catch. I updated the filter to treat `content: null` and empty metadata-only fields like `refusal: null` as part of the leaked bootstrap case, while still rejecting any chunk that carries active content/tool data. I also updated the regression test to cover the `content: null` + `refusal: null` variant.

**Comment by diegosouzapw** [comment]
Merged into `release/v3.8.8` 🎉 Thanks @CitrusIce! Clean fix + regression test (synced with the current release). Ships next release.
